labs/2-perpetual-exchange/3-jlp-pool.md

```diff
@@ -0,0 +1,117 @@
+
+Similarly when withdrawing from a custody, LPs may request to receive a different token than the custody's token. We will use Jupiter Swap to swap the custody's token to the LP's desired token.
+
+And at the point of depositing assets into JLP pool, the protocol will price in the USD value.
+
+
+#### How to Add Liquidity to JLP:
+
+1. Begin by selecting the token asset you wish to deposit into the JLP pool from the list of token assets available in your wallet.
+2. JLP Pool will automatically identify which token asset in the JLP pool has the lowest weightage and swap your selected asset into that pool (if your chosen output token is not in the list of token assets). Your assets will then be deposited into the pool.
+3. Keep in mind that when you deposit into the JLP pool with a relatively high weightage, fees will be incurred. Conversely, depositing into a token pool with lower weightage will result in lower or even zero fees during the deposit process.
+4. For instance, if users deposit a token asset that is not in the pool (e.g., SOL, ETH, BTC, USDC, or USDT), swap will be executed into the token asset with the lowest weightage, ensuring minimal fees.
+
+![JLP1](../img/jlp1.jpg)
+
+#### How to Remove Liquidity from JLP:
+
+1. Start by selecting the token asset you intend to withdraw from the JLP pool. You can choose from the list of available token assets in the pool, and the chosen asset will be withdrawn directly into your wallet.
+2. JLP Pool will automatically identify the token asset in the JLP pool with the highest weightage and swap your chosen asset into that pool if your selected output token is not in the list of token assets. Your assets will then be withdrawn directly into your wallet.
+3. It's important to note that when you withdraw from the JLP token pool with a relatively low weightage, fees will be incurred. Conversely, withdrawing from a token pool with higher weightage will result in lower or even zero fees during the withdrawal process.
+4. In situations where users are withdrawing a token asset not present in the pool (e.g., SOL, ETH, BTC, USDC, or USDT), JLP will withdraw from the token pool with the highest weightage, incurring minimal fees. The withdrawn assets will be swapped into the selected output token and then transferred directly into the user's wallet.
+
+![JLP2](../img/jlp2.jpg)
+
+### Target Ratio and Fees
+
+In the JLP pool, every token holds a specific target ratio or weightage.
+
+The transactions involving the addition or removal of liquidity have the primary purpose of adjusting a token's ratio in the pool to align it more closely with the predefined target.
+
+Similarly, each custody plays a role in maintaining a target ratio within the pool. This target ratio significantly influences the fees that are applicable to instructions aimed at modifying the ratio of the custodies within the pool.
```

Comment:
> Similarly, each custody plays a role in maintaining a target ratio within the pool.

This sentence can be removed without affecting the info as it is implementation detail.

> This target ratio significantly ...

This is the key part but this sentence is a bit convoluted. One possible alternative. "Instructions that move the token's ratio away from the target incurs additional fees while instructions that move it closer to the target get a fee rebate."

Reply:
Should we use the word `fee rebate`?

Reply:
Or fee discount?

Reply:
either should be fine, discount seems easier to understand